cmd/kubeadm/app/node/bootstrap.go

```diff
@@ -105,6 +105,23 @@ func EstablishMasterConnection(s *kubeadmapi.NodeConfiguration, clusterInfo *kub
 	return establishedConnection, nil
 }
 
+// check to see if there are other nodes in the cluster with identical node names.
+func CheckForNodeNameDuplicates(connection *ConnectionDetails) error {
+	hostName, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("<node/bootstrap> failed to get node hostname [%v]", err)
+	}
+	// TODO(phase1+) https://github.com/kubernetes/kubernetes/issues/33641
+	nodeName := types.NodeName(hostName)
+	nodeList, err := connection.ClientSet.Nodes().List(v1.ListOptions{})
+	for _, node := range nodeList.Items {
+		if nodeName == types.NodeName(node.Name) {
```

Comment:
Why convert `node.Name` here when you can just directly check it against `hostName` like this?
```go
if hostName == node.Name
```

I think that's a safer approach